tagstudio/src/core/palette.py

```diff
@@ -13,7 +13,7 @@ class ColorType(int, Enum):
     DARK_ACCENT = 4
 
 
-_TAG_COLORS = {
+_TAG_COLORS: dict = {
```

Comment:
`defaultdict` seems like it's well-suited for this, and would indeed replace the need for the try/except block in `get_tag_color`. However, since #332 is doing [some refactor work](https://github.com/TagStudioDev/TagStudio/blob/f436b5c7a1dd989f65173f4bd560ea3e0fbcfa49/tagstudio/src/core/palette.py) in `palette.py`, I feel it would be best to wait until that one is merged to consider switching to `defaultdict`.

[Custom user-defined tag colors](https://github.com/TagStudioDev/TagStudio/pull/372#issuecomment-2307658099) is also a feature I've been wanting to add in the near future (post-#332) that would see this getting shaken up once more. Still, it's good to be aware of `defaultdict` now!